sympy/functions/elementary/complexes.py

```diff
@@ -92,10 +107,35 @@ def eval(cls, arg):
     def as_real_imag(self, deep=True, **hints):
         """
         Returns the real number with a zero imaginary part.
+
+        Examples
+        ========
+
+        >>> from sympy import re, I
+        >>> from sympy.abc import x, y
+        >>> re(10 + 20*I).as_real_imag()
+        (10, 0)
+        >>> re(10).as_real_imag()
+        (10, 0)
+        >>> re(40*I).as_real_imag()
+        (0, 0)
+        >>> re(x+y*I).as_real_imag()
+        (re(x) - im(y), 0)
+
         """
         return (self, S.Zero)
 
     def _eval_derivative(self, x):
+        '''
+        Examples
+        ========
```

Comment:
We don't need to document this internal function. A demonstration of these capabilities could go in the main class docstring.

Reply:
Okay

Reply:
So examples of the internal functions will be inside the main class docstring. Noted.

Reply:
If there is an example for this it should use `diff` rather than `_eval_derivative`. Users shouldn't call this method directly.